testutil-server/src/main/java/io/spine/testing/server/blackbox/SubscriptionFixture.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2020, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.testing.server.blackbox;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.truth.extensions.proto.IterableOfProtosSubject;
+import com.google.common.truth.extensions.proto.ProtoTruth;
+import io.grpc.stub.StreamObserver;
+import io.spine.base.EntityState;
+import io.spine.base.EventMessage;
+import io.spine.client.Subscription;
+import io.spine.client.SubscriptionUpdate;
+import io.spine.client.Topic;
+import io.spine.server.BoundedContext;
+import io.spine.server.SubscriptionService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.spine.client.SubscriptionUpdate.UpdateCase.ENTITY_UPDATES;
+import static io.spine.client.SubscriptionUpdate.UpdateCase.EVENT_UPDATES;
+
+/**
+ * Allows to assert updates received on a subscription.
+ */
+public final class SubscriptionFixture {
+
+    private final BoundedContext context;
+    private final Topic topic;
+    private final List<SubscriptionUpdate> updates = new ArrayList<>();
```

Comment:
Let's make is a synchronized `List`. Not a `Concurrent` one.

The reason is that the subscription updates in tests may arrive from different threads. The `Concurrent` collections are "eventually consistent", so we want to enable the plain old synchronization.